packages/angular/projects/ui-angular/src/lib/primitives/amplify-input/amplify-input.component.ts

```diff
@@ -40,8 +40,21 @@ export class AmplifyInputComponent {
     return validationError[this.name];
   }
 
+  inferLabel(): string {
+    return this.label || this.attributeMap[this.name]?.label;
+  }
+
+  inferPlaceholder(): string {
+    console.log(this.attributeMap[this.name]);
+    return (
+      this.placeholder ||
+      this.attributeMap[this.name]?.placeholder ||
+      this.inferLabel()
+    );
+  }
```

Comment:
Nice and DRY!

Reply:
Yup, and this will be cleaned more once Angular starts i18n ✨